core/pkg/ingress/annotations/authtls/main.go

```diff
@@ -30,36 +30,31 @@ const (
 	authTLSSecret = "ingress.kubernetes.io/auth-tls-secret"
 )
 
-// SSLCert returns external authentication configuration for an Ingress rule
-type SSLCert struct {
-	Secret       string `json:"secret"`
-	CertFileName string `json:"certFilename"`
-	KeyFileName  string `json:"keyFilename"`
-	CAFileName   string `json:"caFilename"`
-	PemSHA       string `json:"pemSha"`
+type authTLS struct {
+	certResolver resolver.AuthCertificate
+}
+
+// NewParser creates a new TLS authentication annotation parser
+func NewParser(resolver resolver.AuthCertificate) parser.IngressAnnotation {
+	return authTLS{resolver}
 }
 
 // ParseAnnotations parses the annotations contained in the ingress
 // rule used to use an external URL as source for authentication
-func ParseAnnotations(ing *extensions.Ingress,
-	fn func(secret string) (*SSLCert, error)) (*SSLCert, error) {
-	if ing.GetAnnotations() == nil {
-		return &SSLCert{}, parser.ErrMissingAnnotations
-	}
-
+func (a authTLS) Parse(ing *extensions.Ingress) (interface{}, error) {
 	str, err := parser.GetStringAnnotation(authTLSSecret, ing)
 	if err != nil {
-		return &SSLCert{}, err
+		return nil, err
 	}
 
 	if str == "" {
-		return &SSLCert{}, fmt.Errorf("an empty string is not a valid secret name")
+		return nil, ing_errors.NewLocationDenied("an empty string is not a valid secret name")
 	}
 
 	_, _, err = k8s.ParseNameNS(str)
 	if err != nil {
-		return &SSLCert{}, err
+		return nil, ing_errors.NewLocationDenied("an empty string is not a valid secret name")
 	}
 
-	return fn(str)
+	return a.certResolver.GetAuthCertificate(str)
```

Comment:
I think it makes sense to wrap the error returned from this in a location denied.

This errors if the referenced cert can't be resolved, and that sounds like excellent grounds to deny.